src/helpers/presentation.rs

```diff
@@ -347,6 +363,50 @@ impl<'a> Presentation<'a> {
             .map_err(|err| PresentationError::VadeEvanError(err.to_string()))
     }
 
+    /// Creates a self issued presentation.
+    /// The presentation has no proof.
+    ///
+    /// # Arguments
+    ///
+    /// * `unsigned_credential` - self issued credential (without proof) to be shared in presentation
+    ///
+    /// # Returns
+    /// * `Option<String>` - A `SelfIssuedPresentation` as JSON
+    pub async fn create_self_issued_presentation(
+        &mut self,
+        unsigned_credential_str: &str,
+    ) -> Result<String, PresentationError> {
+        if unsigned_credential_str.contains("proof") {
```

Comment:
What happens, when the credential subject data contains a field named proof? :D

Reply:
makes sense, :) . changed the condition , now checking 
=> if credential is signed and finished then throw error otherwise proceed